pkg/controller/certificates/trigger/policies/gatherer_test.go

```diff
@@ -0,0 +1,336 @@
+								"cert-manager.io/certificate-revision": "1",
+							}),
+						)}, nil
+				})
+			}),
+			wantErr: "multiple CertificateRequest resources exist for the current revision, not triggering new issuance until requests have been cleaned up",
+		},
+		"should return error when the list func returns an error": {
+			givenCert: gen.Certificate("cert-1",
+				gen.SetCertificateUID("uid-1"),
+				gen.SetCertificateSecretName("secret-1"),
+				gen.SetCertificateRevision(1),
+			),
+			mockSecretLister: listers.FakeSecretListerFrom(listers.NewFakeSecretLister(),
+				listers.SetFakeSecretNamespaceListerGet(&corev1.Secret{}, nil),
+			),
+			mockCertificateRequestsLister: mockCertificateRequests("default-unit-test-ns", func(t *testing.T) *listers.FakeCertificateRequestNamespaceLister {
+				f := expectCalled(t, 1)
+				return listers.NewFakeCertificateRequestNamespaceLister().WithList(func(_ labels.Selector) ([]*cmapi.CertificateRequest, error) {
+					f()
+					return nil, fmt.Errorf("error that is not a not_found error")
+				})
+			}),
+			wantErr: "error that is not a not_found error",
+		},
+	}
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			g := &Gatherer{
+				CertificateRequestLister: test.mockCertificateRequestsLister(t),
+				SecretLister:             test.mockSecretLister,
+			}
+
+			got, gotErr := g.DataForCertificate(context.Background(), test.givenCert)
+
+			if test.wantErr != "" {
+				assert.Error(t, gotErr)
+				assert.EqualError(t, gotErr, test.wantErr)
+				return
+			}
+
+			require.NoError(t, gotErr)
+			assert.Equal(t, test.wantRequest, got.CurrentRevisionRequest)
+			assert.Equal(t, test.wantSecret, got.Secret)
+			assert.Equal(t, test.givenCert, got.Certificate, "input cert should always be equal to returned cert")
+		})
+	}
+}
+
+// Creates a mock CertificateRequestLister.
+//
+// We want to use a mock instead of a fake here: the mock makes sure that
+// (1) the lister.CertificateRequests(namespace) has been called with the
+// correct expected namespace and (2) lister.CertificateRequests(namespace)
+// has been called exactly once, which makes sure (1) was checked.
+func mockCertificateRequests(expNamespace string, innerLister func(t *testing.T) *listers.FakeCertificateRequestNamespaceLister) func(*testing.T) *listers.FakeCertificateRequestLister {
+	return func(t *testing.T) *listers.FakeCertificateRequestLister {
+		f := expectCalled(t, 1)
+		return listers.
+			NewFakeCertificateRequestLister().
+			WithCertificateRequests(func(namespace string) cmlist.CertificateRequestNamespaceLister {
+				f()
+				assert.Equal(t, expNamespace, namespace)
+				return innerLister(t)
+			})
+	}
+}
+
+// The returned function f is expected to be called expectedCount times.
+// Returns a friendly error with the location where f was created to help
+// the developer figure out where the error came from.
+func expectCalled(t *testing.T, expectedCount uint32) (f func()) {
```

Comment:
Why are we using uint32 here? We should generally use `int` unless `uint` is needed.